docs/content/concepts/cryptography/zklogin.mdx

```diff
@@ -527,7 +527,7 @@ You might want to cache the ephemeral key pair along with the ZKP for future use
 
 However, the ephemeral key pair needs to be treated as a secret akin to a key pair in a traditional wallet. This is because if both the ephemeral private key and ZK proof are revealed to an attacker, then they can typically sign any transaction on behalf of the user (using the same process described previously).
 
-Consequently, you should not store them persistently in an unsecure storage location, on any platform. For example, on browsers, use session storage instead of local storage to store the ephemeral key pair and the ZK proof.
+Consequently, you should not store them persistently in an insecure storage location, on any platform. For example, on browsers, use session storage instead of local storage to store the ephemeral key pair and the ZK proof.
```

Comment:
LGTM, though if you want to further improve the wording, imo there should be more detail here - users could lose funds if this is implemented badly.
Anything written to local storage will usually stay there until the user manually clears their browser cache (so, forever?), whereas data written to session storage gets erased when the browser session ends.

ChatGPT suggested this alternative wording but I think it can be improved:


```
Instead of utilizing local storage, which persists data indefinitely and poses a security risk, opt for session storage for storing ephemeral key pairs and ZK proofs. Session storage is considered more secure for temporary data because it automatically clears the information when the browser session ends, reducing the vulnerability to data breaches.
```

Reply:
```suggestion
Consequently, you should choose the storage of the ephemeral key pair and the ZK proof carefully. For example, a web browser provides both session and local storage. Data in local storage persists until you explicitly clear your browser cache (remains across browser sessions), whereas session storage persists only until you close the tab or bowser. Using local storage for this data is preferable to safeguard against the loss of funds should the session end unexpectedly.
```
I think this breaks up the middle sentence into smaller chunks and keeps your same thought. Your sentence is grammatically correct, but might be difficult for ESL to process. 

As an aside, I think I'd still prefer "unsecure" over "insecure" even though spellcheck flags it. Unsecure is getting more validity from online dictionaries for use in the tech space and doesn't have the mental health connotations that "insecure" has.